fluent/fluent.go

```diff
@@ -347,13 +353,25 @@ func (f *Fluent) connect() (err error) {
 }
 
 func (f *Fluent) run() {
+	drainEvents := false
 	for {
+		select {
+		case stopRunning, ok := <-f.stopRunning:
+			if stopRunning || !ok {
+				drainEvents = true
+			}
+default:
+		}
 		select {
 		case entry, ok := <-f.pending:
 			if !ok {
 				f.wg.Done()
 				return
 			}
+			if drainEvents {
+				fmt.Fprintf(os.Stderr, "[%s] Unable to send logs to fluentd, discarding...\n", time.Now().Format(time.RFC3339))
```

Comment:
This feature is for the case when the destination can't receive events, so the buffer should contain so many entries. If the buffer has many entries to be sent, this message will be shown so many times (and log flood will happen).
Can you update the code to show this message just once (or few times)?

Reply:
Yes, no problem